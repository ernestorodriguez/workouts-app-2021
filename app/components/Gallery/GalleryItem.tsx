import React, {ReactElement, useState} from "react";
import { NavLink } from "react-router-dom";
import { Placeholder } from "semantic-ui-react";

export interface GalleryItemProps {
  id: number;
  name: string;
  alias: string;
  startDate: string;
  thumbnailMedium: string;
  thumbnailDefault: string;
  thumbnailHigh: string;
  category: string;
}

const GalleryItem = ({
  id,
  name,
  alias,
  startDate,
  thumbnailDefault,
  thumbnailMedium,
  thumbnailHigh,
  category,
}: GalleryItemProps): ReactElement => {
  const [loading, setLoading] = useState(true);
  return (
    <article className="gallery-item" data-js="gallery-item" data-id={id}>
      <NavLink className="gallery-item_link" exact to={`/workouts/${alias}`}>
        <div className="gallery-item__image-container">
          <img
            onLoad={() => {
              setLoading(false);
            }}
            srcSet={`${thumbnailMedium} 480w, ${thumbnailHigh} 800w`}
            sizes="(max-width: 600px) 480px, 800px"
            loading="lazy"
            className={`gallery-item__image ${
              loading && "gallery-item__image--loading"
            }`}
            src={thumbnailDefault}
            alt=""
          />
          {loading && (
            <Placeholder>
              <Placeholder.Image rectangular />
            </Placeholder>
          )}
        </div>

        <div className="gallery-item_info">
          <div className="gallery-item_info-name">{name}</div>
          <div data-js="start-date">{startDate}</div>
          <div className="gallery-item__category" data-js="start-category">
            {category}
          </div>
        </div>
      </NavLink>
    </article>
  );
}

export default GalleryItem;
