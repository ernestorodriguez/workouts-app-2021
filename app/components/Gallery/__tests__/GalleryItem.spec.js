import React from 'react';
import { shallow } from 'enzyme';
import GalleryItem from '../GalleryItem';

describe('GalleryItem component', () => {
  test('should match snapshot', () => {
    const props = {
      id: 1,
      name: 'workout-1',
      alias: 'alias-path',
      startDate: '2020-10-24',
      thumbnailMedium: 'http://google.com',
      category: 'c1',
    };

    // eslint-disable-next-line react/jsx-props-no-spreading
    const wrapper = shallow(<GalleryItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
