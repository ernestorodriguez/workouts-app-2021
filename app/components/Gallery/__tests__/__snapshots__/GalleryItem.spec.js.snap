// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItem component should match snapshot 1`] = `
<article
  className="gallery-item"
  data-id={1}
  data-js="gallery-item"
>
  <NavLink
    className="gallery-item_link"
    exact={true}
    to="/workout/alias-path"
  >
    <img
      alt=""
      className="gallery-item__image"
      src="http://google.com"
    />
    <div
      className="gallery-item_info"
    >
      <div
        className="gallery-item_info-name"
      >
        workout-1
      </div>
      <div
        data-js="start-date"
      >
        2020-10-24
      </div>
      <div
        data-js="start-category"
      >
        c1
      </div>
    </div>
  </NavLink>
</article>
`;
